server/index.js

```diff
@@ -143,9 +143,48 @@ app.get('/api/random', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.post('/api/history', (req, res, next) => {
+  let { userId, dateId } = req.body;
+  userId = parseInt(userId);
+  dateId = parseInt(dateId);
+  if (!Number.isInteger(dateId) || !Number.isInteger(userId)) {
+    throw new ClientError(400, 'Both userId and dateId need to be positive integers.');
+  }
+  const sql = `
+    insert into "history" ("dateId", "userId")
+    values ($1, 1)
+    returning "dateId", "addedAt"
+  `;
+  const params = [dateId];
+  db.query(sql, params)
+    .then(result => {
+      res.status(201).json(result.rows[0]);
+    })
+    .catch(err => next(err));
+});
+
+app.get('/api/history', (req, res, next) => {
+  const sql = `
+    select "dates"."dateIdea", "lists"."listTitle", "dates"."dateId", "history"."addedAt"
+    from "dates"
+    join "history" using("dateId")
+    join "lists" using("listId")
+    where "history"."userId" = 1
+    order by "addedAt" desc
+  `;
+  db.query(sql)
+    .then(result => {
+      res.status(200).json(result.rows);
+    })
+    .catch(err => next(err));
+});
+
 app.use(errorMiddleware);
 
 app.listen(process.env.PORT, () => {
   // eslint-disable-next-line no-console
   console.log(`express server listening on port ${process.env.PORT}`);
 });
+
+// .env db URL should read the following, when it's pushed live:
+// DATABASE_URL=postgres://uyxsnztfvdnpam:f08c4af64e29d655b22f265638d7317965a6bb62b773846dc9198d7ebf49dce9@ec2-35-174-56-18.compute-1.amazonaws.com:5432/d27uu2jroi6rdi
```

Comment:
Bro, you just published your private database url. You need to destroy the one you have, delete this line and then have Heroku attach a new one.

Reply:
My mistake.
I've destroyed this URL and attached a new, private one